src/main/java/billgates/view/gui/AddEntryDialog.java

```diff
@@ -0,0 +1,107 @@
+package billgates.view.gui;
+
+import com.github.lgooddatepicker.components.DateTimePicker;
+
+import javax.swing.*;
+import java.awt.*;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+public class AddEntryDialog extends JDialog {
+
+    private int result;
+    private ZonedDateTime date;
+    private static double value;
+    private String currency;
+    private String description;
+    private String from;
+    private String to;
+    private String location;
+    private final MainFrame mainFrame;
+
+    public AddEntryDialog(MainFrame owner) {
+        super(owner);
+        this.mainFrame = owner;
+        // Create a datetime picker.
+        DateTimePicker dateTimePicker1 = new DateTimePicker();
+        add(dateTimePicker1);
+
+        JTextField valueField = new JTextField(20);
+        valueField.setDocument(new RegexDocument("\\S*"));
+        valueField.setDocument(new RegexDocument("\\d*\\.?\\d{0,2}"));
+
+        JTextField currencyField = new JTextField(20);
+        currencyField.setDocument(new RegexDocument("\\S*"));
+        currencyField.setDocument(new RegexDocument("[A-Z]*"));
```

Comment:
Remember that currency should only be 3 character length.
However, we can keep it just for now.